packages/compass-signtool/src/signing-clients/remote-signing-client.ts

```diff
@@ -0,0 +1,109 @@
+import path from 'path';
+import type { SFTPWrapper } from 'ssh2';
+import type { SSHClient } from '../ssh-client';
+import { debug } from '../utils';
+import type { SigningClient, SigningClientOptions } from '.';
+
+export class RemoteSigningClient implements SigningClient {
+  private sftpConnection!: SFTPWrapper;
+
+  constructor(
+    private sshClient: SSHClient,
+    private options: SigningClientOptions
+  ) {}
+
+  /**
+   * Initialize the signing client and setup remote machine to be ready for signing
+   * the files. This will do following things:
+   * - Create a working directory on the remote machine
+   * - Copy the signing script to the remote machine
+   */
+  private async init() {
+    this.sftpConnection = await this.sshClient.getSftpConnection();
+    await this.sshClient.exec(`mkdir -p ${this.options.rootDir}`);
```

Comment:
Here I guess we can't do fs calls instead, but let's at least try to do some minimal escaping?
```suggestion
    await this.sshClient.exec(`mkdir -p '${this.options.rootDir}'`);
```